client/src/main/java/io/pravega/client/stream/notifications/NotifierFactory.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.stream.notifications;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+
+import io.pravega.client.state.StateSynchronizer;
+import io.pravega.client.stream.impl.ReaderGroupState;
+import io.pravega.client.stream.notifications.notifier.ScaleEventNotifier;
+import lombok.Synchronized;
+
+/**
+ * Factory used to create different types of notifiers.
+ */
+public class NotifierFactory {
+
+    private final NotificationSystem system;
+    private ScaleEventNotifier scaleEventNotifier;
+
+    public NotifierFactory(final NotificationSystem notificationSystem) {
+        this.system = notificationSystem;
+    }
+
+    @Synchronized
```

Comment:
Why does this method need to be synchronized?

Reply:
ScaleEventNotifier is a singleton and there should not be more than one instance of this object